test/runtests.jl

```diff
@@ -1446,7 +1729,82 @@ end
     end
 end
 
-@testset "dropcollinear with GLMs" begin
+@testset "contrasts argument" begin
+    # DummyCoding (default)
+    m = lm(@formula(y~x), (y=1:25, x=repeat(1:5, 5)),
+           contrasts=Dict(:x=>DummyCoding()))
+    mcat = lm(@formula(y~x), (y=1:25, x=categorical(repeat(1:5, 5))))
+    gm = glm(@formula(y~x), (y=1:25, x=repeat(1:5, 5)), Normal(), IdentityLink(),
+             contrasts=Dict(:x=>DummyCoding()))
+    gmcat = glm(@formula(y~x), (y=1:25, x=categorical(repeat(1:5, 5))),
+                Normal(), IdentityLink())
+    @test coef(m) == coef(mcat) ≈ [11, 1, 2, 3, 4]
+    @test coef(gm) == coef(gmcat) ≈ [11, 1, 2, 3, 4]
+
+    m = fit(LinearModel, @formula(y~x), (y=1:25, x=repeat(1:5, 5)),
+           contrasts=Dict(:x=>DummyCoding()))
+    mcat = fit(LinearModel, @formula(y~x), (y=1:25, x=categorical(repeat(1:5, 5))))
+    gm = fit(GeneralizedLinearModel, @formula(y~x),
+             (y=1:25, x=repeat(1:5, 5)), Normal(), IdentityLink(),
+             contrasts=Dict(:x=>DummyCoding()))
+    gmcat = fit(GeneralizedLinearModel, @formula(y~x),
+                (y=1:25, x=categorical(repeat(1:5, 5))),
+                Normal(), IdentityLink())
+    @test coef(m) == coef(mcat) ≈ [11, 1, 2, 3, 4]
+    @test coef(gm) == coef(gmcat) ≈ [11, 1, 2, 3, 4]
+
+    # EffectsCoding
+    m = lm(@formula(y~x), (y=1:25, x=repeat(1:5, 5)),
+           contrasts=Dict(:x=>EffectsCoding()))
+    gm = glm(@formula(y~x), (y=1:25, x=repeat(1:5, 5)), Normal(), IdentityLink(),
+             contrasts=Dict(:x=>EffectsCoding()))
+    @test coef(m) ≈ coef(gm) ≈ [13, -1, 0, 1, 2]
+
+    m = fit(LinearModel, @formula(y~x), (y=1:25, x=repeat(1:5, 5)),
+           contrasts=Dict(:x=>EffectsCoding()))
+    gm = fit(GeneralizedLinearModel, @formula(y~x),
+             (y=1:25, x=repeat(1:5, 5)), Normal(), IdentityLink(),
+             contrasts=Dict(:x=>EffectsCoding()))
+    @test coef(m) ≈ coef(gm) ≈ [13, -1, 0, 1, 2]
+end
+
+
+@testset "dofit argument" begin
+    gm1 = glm(@formula(y~x), (y=1:25, x=repeat(1:5, 5)), Normal(), IdentityLink(),
+              dofit=true).model
+    gm2 = glm(@formula(y~x), (y=1:25, x=repeat(1:5, 5)), Normal(), IdentityLink(),
+              dofit=false).model
+    @test gm1.fit
+    @test !gm2.fit
+    fit!(gm2)
+    @test gm2.fit
+    @test coef(gm1) == coef(gm2) ≈ [10, 1]
+
+    # Deprecated
+    gm1 = fit(GeneralizedLinearModel, @formula(y~x),
+             (y=1:25, x=repeat(1:5, 5)), Normal(), IdentityLink(),
+             dofit=true).model
+    gm2 = fit(GeneralizedLinearModel, @formula(y~x),
+             (y=1:25, x=repeat(1:5, 5)), Normal(), IdentityLink(),
+             dofit=false).model
+    @test gm1.fit
+    @test !gm2.fit
+    fit!(gm2)
+    @test gm2.fit
+    @test coef(gm1) == coef(gm2) ≈ [10, 1]
+end
+
+@testset "formula accessor" begin
```

Comment:
This accessor isn't introduced by the PR, right? Why are tests added then?

Reply:
They were part of the cherry-picked commit.

Reply:
I don't see it at https://github.com/JuliaStats/GLM.jl/pull/507/files. I think this comes from the commit which removed `TableRegressionModel`, which isn't relevant here.